pandas/core/indexes/base.py

```diff
@@ -5968,14 +5996,24 @@ def map(self, mapper, na_action=None):
             # empty
             dtype = self.dtype
 
+        # e.g. if we are floating and new_values is all ints, then we
+        #  don't want to cast back to floating.  But if we are UInt64
+        #  and new_values is all ints, we want to try.
```

Comment:
Not necessarily needed for this PR, but I for such a case of floating index with new_values being int, we ideally preserve the fact that it is a nullable dtype.

Reply:
requested action here or can i hit "resolve conversation"?

Reply:
Requested action is either fix here or note somewhere as a follow-up (probably the second)